feat(theme-common): code block MagicComments support for TeX/LaTeX/Matlab

jj-style opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.
You can learn more about contributing to Docusaurus here: https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md
Happy contributing!
-->

## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

<!--
Please also remember to sign the CLA, although you can also sign it after submitting the PR. The CLA is required for us to merge your PR.
If this PR adds or changes functionality, please take some time to update the docs. You can also write docs after the API design is finalized and the code changes have been approved.
-->

## Motivation

<!-- Help us understand your motivation by explaining why you decided to make this change. Does this fix a bug? Does it close an issue? -->
I could not see an open issue for it but I would like to be able to use the magic comments feature in Matlab code blocks. This isn't currently supported due to Matlab's way of commenting, therefore I have added this capability in this PR.

## Test Plan

<!-- Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos! -->
- I have added a Matlab code snippet in the
 dogfooding page which verifies the functionality.
- I ran a local instance of the site and verified that the lines were not highlighted by the magic comments
- I did the implementation and restarted the server and verified that the lines were highlighted

before
![before](https://github.com/facebook/docusaurus/assets/27488093/94416962-5a97-4b01-a9bd-b6181ef3f9db)
after
![after](https://github.com/facebook/docusaurus/assets/27488093/6d8eb6b8-0af1-4580-b7ab-135369a31e54)





### Test links

<!--
🙏 Please add an exhaustive list of links relevant to this pull request.
⏱ This saves maintainers a lot of time during reviews.

- If you changed anything that's displayed on UI, please add a dogfooding page in website/_dogfooding to help us preview the effect. Those tests are deployed at https://docusaurus.io/tests
- If you changed documentation, please link to the new and updated documentation pages.

After submission, our Netlify bot will post a deploy preview link in comment, in the format of https://deploy-preview-<PR-NUMBER>--docusaurus-2.netlify.app/. Once available, please edit this section with links to the relevant deploy preview pages.

Please don't be afraid to change the main site's configuration as well! You can make use of your new feature on our site so we can preview its effects. We can decide if it should be kept in production before merging it.
-->

Deploy preview: https://deploy-preview-8982--docusaurus-2.netlify.app/tests/pages/code-block-tests/#magic-comments-tests

## Related issues/PRs

<!-- If you haven't already, link to issues/PRs that are related to this change. This helps us develop the context and keep a rich repo history. If this PR is a continuation of a past PR's work, link to that PR. If the PR addresses part of the problem in a meta-issue, mention that issue. -->

Josh-Cena reviewed (commented):
Thanks! This pattern is inherited from TeX so let's call the style `tex` and add `tex` and `latex` to the language list.

jj-style commented:
Thank you for the feedback!
I have now changed the comment style to be called `tex` and added an example in dogfooding code blocks page to verify this for `latex` also

slorber commented on website/_dogfooding/_pages tests/code-block-tests.mdx:
````diff
@@ -322,3 +322,29 @@ WHERE customer_id IN (
+y = times2(x);
+```
+
+```latex title="latex.tex
````
```suggestion
```latex title="latex.tex"
```

Missing close quote

jj-style commented:
Thanks for the review and for pushing up the missing quote! 😊
